### PR TITLE
When the texture is not rotated, the rendering fails

### PR DESCRIPTION
**Pull request by jorpiell**
Combine the patch that @dwhipps suggested to fix the bug #2737 and #4515

### CODE REVIEW

**Comment by dwhipps** [comment]
@jorpiell This is great, but any idea WHY this is true? I feel like it might be similar to the bug @hpinkos found here [https://github.com/AnalyticalGraphicsInc/cesium/pull/4206/commits/564dbb86c5a4cd946f34a8c2211cc02a3162c8d4](https://github.com/AnalyticalGraphicsInc/cesium/pull/4206/commits/564dbb86c5a4cd946f34a8c2211cc02a3162c8d4)

File: Source/Core/RectangleGeometry.js [hunk 1]
```diff
@@ -802,6 +802,7 @@ define([
         options.textureMatrix = textureMatrix;
         options.tangentRotationMatrix = tangentRotationMatrix;
         options.size = options.width * options.height;
+        options.stRotation = stRotation;
```
Comment (mramato) [inline Source/Core/RectangleGeometry.js]: Is this actually used anywhere?  I traced through the code visually and didn't see this particular options object including `stRotation`.

**Comment by mramato** [comment]
@bagnell and @hpinkos can you take a look at this?  Obviously the rotation issues are a hot topic so it would be nice if we can track down the root cause and get them fixed.

File: Source/Core/RectangleGeometry.js (hunk 1 above)
Comment (hpinkos) [inline Source/Core/RectangleGeometry.js]: Yes, it's used to make a `textureMatrix` that rotates the st coordinates

Comment (mramato) [inline Source/Core/RectangleGeometry.js]: Sorry to be dense, but where is that happening?  From what I can tell, only `constructRectangle` and `constructExtrudedRectangle` get called with this specific options object and neither of them use it.

Comment (mramato) [inline Source/Core/RectangleGeometry.js]: Nevermind, I see it now.  Sorry for the noise.

**Comment by dwhipps** [comment]
@hpinkos I've always been told that the squeaky wheel gets the grease... here is my weekly squeak for this one.

**Comment by hpinkos** [comment]
I didn't forget about this @jorpiell, @dwhipps 
I'm not sure if this is the best solution, but I'm going to keep this PR open until I can dig into it a little more.
We have a code sprint coming up the first week of December, I'll definitely look at it then if I don't get to it sooner

**Comment by hpinkos** [comment]
@jorpiell See the fix I submitted here: #4725

**Pull request closed without merging**